### PR TITLE
WIP(llgo/rust/hyper): add a hyper client demo

### PR DESCRIPTION
**Pull request by spongehah**
Now run the error, try using c-rust socket。

(base) spongehah@MacBook hyper % llgo run .
panic: compileInstrAndValue: unknown instr - *ssa.SliceToArrayPointer

goroutine 1 [running]:
github.com/goplus/llgo/cl.(*context).compileInstrOrValue(0xc0066b1450, 0xc0077362d0, {0x1092b8f18?, 0xc00ccb96d0?}, 0x68?)
        /Users/spongehah/go/src/llgo/cl/compile.go:604 +0x124b
github.com/goplus/llgo/cl.(*context).compileInstr(0xc0066b1450, 0x107266e01?, {0x100478948?, 0xc00ccb96d0?})
        /Users/spongehah/go/src/llgo/cl/compile.go:618 +0x665
github.com/goplus/llgo/cl.(*context).compileBlock(0xc0066b1450, 0xc0077362d0, 0xc00cd01340, 0x100479a18?, 0x0, 0x0)
        /Users/spongehah/go/src/llgo/cl/compile.go:311 +0x51d
github.com/goplus/llgo/cl.(*context).compileFuncDecl.func1()
        /Users/spongehah/go/src/llgo/cl/compile.go:259 +0x3ef
github.com/goplus/llgo/cl.NewPackageEx(0xc00014e000, 0xc0082c63f0, 0xc00c883e00, {0xc0063b7698, 0x3, 0x3})
        /Users/spongehah/go/src/llgo/cl/compile.go:832 +0x691
github.com/goplus/llgo/internal/build.buildPkg(0xc00b557630, 0xc00cfdcd80, 0x0)
        /Users/spongehah/go/src/llgo/internal/build/build.go:432 +0x308
github.com/goplus/llgo/internal/build.buildAllPkgs(0xc00b557630, {0xc003e2a478?, 0xc00779e698?, 0xd?}, 0xd?)
        /Users/spongehah/go/src/llgo/internal/build/build.go:301 +0x455
github.com/goplus/llgo/internal/build.Do({0xc000024170?, 0x1?, 0x1?}, 0xc00007b450)
        /Users/spongehah/go/src/llgo/internal/build/build.go:178 +0x934
github.com/goplus/llgo/cmd/internal/run.runCmdEx(0xc00013de48?, {0xc000024170?, 0x1000fef00?, 0x3?}, 0x10040550f?)
        /Users/spongehah/go/src/llgo/cmd/internal/run/run.go:62 +0xb0
github.com/goplus/llgo/cmd/internal/run.runCmd(0xc000126000?, {0xc000024170?, 0x0?, 0xc00004c6b8?})
        /Users/spongehah/go/src/llgo/cmd/internal/run/run.go:50 +0x1d
main.main()
        /Users/spongehah/go/src/llgo/cmd/llgo/llgo.go:90 +0x4b3

### CODE REVIEW

**Comment by spongehah** [comment]
Run the following command in hyper to generate a dylib file:
```
RUSTFLAGS="--cfg hyper_unstable_ffi" cargo rustc --features client,http1,http2,ffi --crate-type cdylib
```

File: rust/hyper/hyper.go [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+	_ "unsafe"
+)
+
+// Write the .pc file for the dylib generated by the Rust library and copy it to pkg-config for proper location.
+const (
+	LLGoPackage = "link: $(pkg-config --libs hyper_client); -lhyper"
+)
+
+const (
+	HYPER_ITER_CONTINUE     = 0
```
Comment (xushiwei) [inline rust/hyper/hyper.go]: remove prefix `HYPER_`

File: rust/hyper/hyper.go [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+
+const (
+	HYPER_TASK_EMPTY = iota
+	HYPER_TASK_ERROR
+	HYPER_TASK_CLIENTCONN
+	HYPER_TASK_RESPONSE
+	HYPER_TASK_BUF
+)
+
+type Hyper_task_return_type struct {
```
Comment (xushiwei) [inline rust/hyper/hyper.go]: rename `Hyper_task_return_type` => `TaskReturnType`

File: rust/hyper/hyper.go [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+	Unused [8]byte
+}
+
+type Hyper_waker struct {
+	Unused [8]byte
+}
+
+// Returns a static ASCII (null terminated) string of the hyper_client version.
+// llgo:link Hyper_version C.hyper_version
+func Hyper_version() *c.Char { return nil }
```
Comment (xushiwei) [inline rust/hyper/hyper.go]: rename `Hyper_version` => `Version`

File: rust/hyper/hyper.go [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+
+// Creates a new "empty" body.
+// llgo:link Hyper_body_new C.hyper_body_new
+func Hyper_body_new() *Hyper_body {
```
Comment (xushiwei) [inline rust/hyper/hyper.go]: rename `Hyper_body_new` => `NewBody`

File: rust/hyper/hyper.go [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+	return nil
+}
+
+// Free a body.
+// llgo:link (*Hyper_body).Hyper_body_free C.hyper_body_free
+func (body *Hyper_body) Hyper_body_free() {
```
Comment (xushiwei) [inline rust/hyper/hyper.go]: rename `Hyper_body_free` => `Free`

File: rust/hyper/hyper.go (hunk 1 above)
Comment (spongehah) [reply rust/hyper/hyper.go]: All changes have been made

**Pull request merged**